optional_plugins/golang/avocado_golang/__init__.py

```diff
@@ -0,0 +1,225 @@
+"""
+
+import fnmatch
+import glob
+import os
+import re
+
+from avocado.core import loader
+from avocado.core import output
+from avocado.core import test
+from avocado.core.plugin_interfaces import CLI
+from avocado.utils import path as utils_path
+from avocado.utils import process
+
+
+_GO_BIN = utils_path.find_command('go')
+
+
+class GolangTest(test.SimpleTest):
+
+    """
+    Run a Golang Test command as a SIMPLE test.
+    """
+
+    def __init__(self, name,
+                 params=None,
+                 base_logdir=None,
+                 job=None,
+                 subtest=None):
+        super(GolangTest, self).__init__(name, params, base_logdir, job)
+        self.subtest = subtest
+
+    @property
+    def filename(self):
+        """
+        Returns the path of the golang test suite.
+        """
+        return self.name.name.split(':')[0]
+
+    def test(self):
+        """
+        Create the Golang command and execute it.
+        """
+        test_name = '%s$' % self.name.name.split(':')[1]
+        if self.subtest is not None:
+            test_name += '/%s' % self.subtest
+
+        cmd = '%s test -v %s -run %s' % (_GO_BIN, self.filename, test_name)
+
+        result = process.run(cmd, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail('Golang Test execution returned a '
+                      'non-0 exit code (%s)' % result)
+
+
+class NotGolangTest(object):
+
+    """
+    Not a golang test (for reporting purposes)
+    """
+
+
+class GolangLoader(loader.TestLoader):
+    """
+    Golang loader class
+    """
+    name = "golang"
+
+    def __init__(self, args, extra_params):
+        super(GolangLoader, self).__init__(args, extra_params)
+
+    def discover(self, url, which_tests=loader.DEFAULT):
+        if url is None:
+            return []
+
+        avocado_suite = []
+        package_paths = []
+        test_files = []
+        subtest = None
+        tests_filter = None
+
+        if ':' in url:
+            url, _tests_filter = url.split(':', 1)
+            parsed_filter = re.split(r'(?<!\\)/', _tests_filter, 1)
+            _tests_filter = parsed_filter[0]
+            if len(parsed_filter) > 1:
+                subtest = parsed_filter[1]
+            tests_filter = re.compile(_tests_filter)
+
+        # When a file is provided
+        if os.path.isfile(url):
+            test_suite = self._find_tests(url)
+            for item in test_suite:
+                test_name = "%s:%s" % (url, item)
+                if tests_filter and not tests_filter.search(test_name):
+                    continue
+                avocado_suite.append((GolangTest, {'name': test_name,
+                                                   'subtest': subtest}))
+
+            return avocado_suite or self._no_tests(which_tests, url)
+
+        # When a directory is provided
+        if os.path.isdir(url):
+            files = self._find_files(url, recursive=False)
+            for file in files:
```

Comment:
`file` is a reserved builtin, please rename the variable... (path/paths is the usual choice)

Reply:
<o> sure sure